tests/test_type_inference/test_tnode_structure.py

```diff
@@ -26,12 +26,12 @@ def _nodes_to_dict(node_list):
     return node_dict
 
 
-def _find_type_fail(ast_node):
-    if isinstance(ast_node.inf_type, TypeFail):
+def find_type_fail(ast_node):
+    if hasattr(ast_node, 'inf_type') and isinstance(ast_node.inf_type, TypeFail):
```

Comment:
Use of `hasattr` is generally a bad sign. If you feel the need to use it, talk to me *before* putting it in.